### PR TITLE
【ITGD-4765】List

### PR DESCRIPTION
**Pull request by dio-chu**
主要 commit 
[feat(ITGD-4765):PR修改，拆分checkbox邏輯，組件僅維持視覺功能](https://github.com/dio-chu/interview-todo-list/pull/8/commits/e02efa879e37266e6de1fcd82af9f5a12fb80a19)
--
[feature:add List Component](https://github.com/dio-chu/interview-todo-list/commit/0f40130f940e8a156ea491f8cf1ce88d569f362f)

組件
src/components/list/List.vue
組件範例
src/pages/tests/ListTest.vue

### CODE REVIEW

**Comment by dio-chu** [comment]
@simpsons01 
2024/9/1

[feat(ITGD-4765):PR修改，拆分checkbox邏輯，組件僅維持視覺功能](https://github.com/dio-chu/interview-todo-list/pull/8/commits/e02efa879e37266e6de1fcd82af9f5a12fb80a19)

File: src/components/list/List.vue [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<template>
+  <div class="list-container">
+    <ListHeader
+      :headers="headers"
+      :needShowCheckbox="needShowCheckbox"
+      :isChecked="isAllSelected"
+      @update:isChecked="$emit('toggle-all', $event)"
+    />
```
Comment (simpsons01) [inline src/components/list/List.vue]: 建議可以開個header slot
```
<slot nam"header" v-bind="{ headers, needShowChecknox, isAllSelected }">
    <ListHeader
      :headers="headers"
      :needShowCheckbox="needShowCheckbox"
      :isChecked="isAllSelected"
      @update:isChecked="$emit('toggle-all', $event)"
    />
</slot>
```

File: src/components/list/List.vue [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      :isSelected="selectedItems[item.id]"
+      :colors="columnColors"
+      @toggleItem="$emit('toggle-item', item.id)"
+    >
+      <template v-for="header in headers" #[header.key]="{ item }">
+        <slot :name="header.key" :item="item">
+          {{ item[header.key] }}
+        </slot>
+      </template>
+    </ListItem>
```
Comment (simpsons01) [inline src/components/list/List.vue]: 同上，可以開個slot讓外層可以抽換ListItem

File: src/components/list/List.vue [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      default: false,
+    },
+    columnColors: {
+      type: Object,
+      default: () => ({}),
+    },
+    selectedItems: {
+      type: Object,
+      default: () => ({}),
+    },
```
Comment (simpsons01) [inline src/components/list/List.vue]: 想討論是否直接讓props.data的Array裡面多個`isSelected`屬性，讓父層決定是否isSelected而不是在元件本身計算是不是isSelected

**Review by simpsons01 — APPROVED** [review]

**Pull request merged**